cheroot/workers/monitor.py

```diff
@@ -0,0 +1,155 @@
+"""Monitor for threadpool."""
+
+import threading
+import time
+
+from .dynpool import DynamicPoolResizer
+
+
+class BackgroundTask(threading.Thread):
+    """A subclass of threading.Thread whose run() method repeats.
+
+    Use this class for most repeating tasks. It uses time.sleep() to wait
+    for each interval, which isn't very responsive; that is, even if you call
+    self.cancel(), you'll have to wait until the sleep() call finishes before
+    the thread stops. To compensate, it defaults to being daemonic, which means
+    it won't delay stopping the whole process.
+    """
+
+    def __init__(
+        self,
+        interval,
+        function,
+        args=None,
+        kwargs=None,
+        logger=None,
+    ):
+        """Initialize a BackgroundTask."""
+        super(BackgroundTask, self).__init__()
+        self.interval = interval
+        self.function = function
+        self.args = args if args else []
+        self.kwargs = kwargs if kwargs else {}
+        self.running = False
+        self.log = logger or (lambda msg: None)
```

Comment:
Not sure why this is needed.

Reply:
Intention: 
Wanted a logger to be passed to the monitor and use that. 
If input logger is not provided add a none logger. 
Added this way as CherryPy seemed to be doing it this way. 

Question: Any specific way I should handle this?

Reply:
As you may notice, there's no standard way to use logging in Cheroot. I'd be interested in adding logging support across the whole Cheroot in a consistent manner. But please don't add it in this PR as it's unrelated to its main goal and debating on how to implement it properly will block the whole effort so this has to be separate.

Reply:
Got it. Removing logging from here.

Reply:
Done.